### PR TITLE
add getFriendsLIst functions to databse + add test for mockdatabse + …

### LINKED ISSUE #218: add-database-function-to-get-live-data-friendslist

### PR DESCRIPTION
**Pull request by HugoHof**
This PR contains:
- getFriendsLIst functions in the database to obtain a live data of the friends list of the user from the database 
- add test to mock dataBase test to test this new function
- adapt the friends fragment to listen changes from database.
Thank you for your review.
Don't hesitate if you have any question.
close #218

### CODE REVIEW

File: app/src/main/java/com/epfl/drawyourpath/mainpage/fragments/helperClasses/FriendsViewModel.kt [hunk 1]
```diff
@@ -16,57 +16,50 @@ class FriendsViewModel(private val userId: String, private val database: Databas
...
+        // Counter for tracking when all friends are loaded
+        var friendsLoaded = 0
+
+        // Iterate through each userId in friendsList
+        for (userId in newFriendsList) {
+            // Get the username CompletableFuture
+            val usernameFuture = database.getUsername(userId)
+
+            // When the CompletableFuture completes, update the list of friends
+            usernameFuture.whenComplete { username, exception ->
```
Comment (HoLuDyEr) [inline app/src/main/java/com/epfl/drawyourpath/mainpage/fragments/helperClasses/FriendsViewModel.kt:36]: It is more concise to directly call the function without creating a val that is not used :
```kotlin
database.getUsername(userId).whenComplete { username, exception ->
```

File: app/src/main/java/com/epfl/drawyourpath/database/Database.kt [hunk 2]
```diff
@@ -276,6 +276,13 @@ abstract class Database {
      */
     abstract fun getChatList(userId: String): LiveData<List<String>>
 
+    /**
+     * Function used to obtained the list of friends of the user with the given userId.
```
Comment (HoLuDyEr) [inline app/src/main/java/com/epfl/drawyourpath/database/Database.kt:280]: typo : used to obtained -> used to obtain

**Review by HoLuDyEr — APPROVED** [review]
Overall this is a good PR. Using liveData is great to have a responsive UI. I already approve as the changes to be done are minor details.

**Comment by HugoHof** [comment]
> Overall this is a good PR. Using liveData is great to have a responsive UI. I already approve as the changes to be done are minor details.

Thank you for the review

**Pull request merged**